src/plugins/storage.ts

```
@@ -0,0 +1,74 @@
+ * - Vercel Blob Storage is included as a fallback
+ * - If neither is configured, PayloadCMS will use its default local storage
+ */
+
+// Check if S3 credentials are available (including Supabase configuration)
+const hasS3Credentials = Boolean(
+  process.env.S3_ACCESS_KEY && process.env.S3_SECRET_KEY && process.env.S3_BUCKET,
+)
+
+// Check if we're using Supabase S3 endpoint
+const isSupabaseS3 = Boolean(
+  process.env.S3_ENDPOINT && process.env.S3_ENDPOINT.includes('supabase.co'),
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)